### PR TITLE
Logcxx-521

### PR DESCRIPTION
**Pull request by rm5248**
Link with ODBC library if found on OSX and Linux.

### CODE REVIEW

File: src/main/include/CMakeLists.txt [hunk 1]
```diff
@@ -67,12 +67,17 @@ include(CheckCXXSymbolExists)
...
-    set(HAS_ODBC 1)
-endif(${odbc_FOUND})
+if(WIN32)
+    CHECK_INCLUDE_FILE(sqlext.h HAS_ODBC)
+else()
+    include(FindPkgConfig)
+
+    pkg_check_modules( odbc QUIET odbc )
+    if(${odbc_FOUND})
+	set(HAS_ODBC 1)
```
Comment (ams-tschoening) [inline src/main/include/CMakeLists.txt:78]: tabs vs. spaces as well.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,12 @@ if(WIN32)
 # The ODBC appender is always enabled in the Windows configuration
 target_link_libraries(log4cxx PRIVATE odbc32.lib)
 option(LOG4CXX_INSTALL_PDB "Install .pdb files (if generated)"  ON)
+else()
+    if(${odbc_FOUND})
+	target_include_directories(log4cxx PRIVATE ${odbc_INCLUDE_DIRS})
+	target_link_directories( log4cxx PRIVATE ${odbc_LIBRARY_DIRS})
+	target_link_libraries( log4cxx PRIVATE ${odbc_LIBRARIES})
```
Comment (ams-tschoening) [inline src/CMakeLists.txt:15]: These lines mixed tabs and spaces.

**Review by ams-tschoening — CHANGES_REQUESTED** [review]
Minor code style changes, but hings built for me as well.

**Comment by rm5248** [comment]
Dang... not sure how those spaces keep creeping in.  I'll look into making a pre-commit hook to check for those, it should help reduce those errors.  Should be fixed now though.

**Review by ams-tschoening — APPROVED** [review]

**Comment by ams-tschoening** [comment]
> Dang... not sure how those spaces keep creeping in.[...]

`.editorconfig` is the root cause most likely, that contains spaces for shell scripts and `CMakeLists.txt`:

```ini
[*.sh]
charset						= utf-8
end_of_line					= lf
indent_size					= 2
indent_style				= space
insert_final_newline		= true
tab_width					= 2
trim_trailing_whitespace	= true

[CMakeLists.txt]
charset						= utf-8
end_of_line					= lf
indent_size					= 2
indent_style				= space
insert_final_newline		= true
tab_width					= 2
trim_trailing_whitespace	= true
```

Looking at the logs, you seem to have simply copied the block for shell scripts, maybe because some of those files didn't follow the other style guide then. In the easiest case it might get removed so that it's handled by `*`.

**Pull request merged**